site/e2e/playwright.config.ts

```diff
@@ -7,13 +8,33 @@ import {
   e2eFakeExperiment1,
   e2eFakeExperiment2,
   gitAuth,
+  requireTerraformTests,
 } from "./constants";
 
 export const wsEndpoint = process.env.CODER_E2E_WS_ENDPOINT;
 
 // This is where auth cookies are stored!
 export const storageState = path.join(__dirname, ".auth.json");
 
+if (requireTerraformTests) {
+  try {
+    // If running terraform tests, verify the requirements exist in the
+    // environment.
+    //
+    // These execs will throw an error if the status code is non-zero.
+    // So if both these work, then we can launch terraform provisioners.
+    execSync("terraform --version");
+    execSync("docker --version");
+  } catch (e) {
+    throw new Error(
+      "Terraform provisioners require docker & terraform. " +
+        "At least one of these is not present in the runtime environment. To check yourself:\n" +
+        "\t$ terraform --version\n" +
+        "\t$ docker --version",
+    );
+  }
```

Comment:
it'd be kind of nice to check these separately and report a specific one as missing (or maybe both)

maybe something like...

```ts
let hasTerraform = false;
let hasDocker = false;

try {
  execSync("terraform --version");
  hasTerraform = true;
} catch {}

try {
  execSync("docker --version");
  hasDocker = true;
} catch {}

if (!hasTerraform || !hasDocker) {
  throw new Error("<something about what specifically is missing>")
}
```

Reply:
It kinda does. It looks like this on a failure:
```
/bin/sh: line 1: docker: command not found
Error: Terraform provisioners require docker & terraform. At least one of these is not present in the runtime environment. To check yourself:
        $ terraform --version
        $ docker --version
```

The line before the error is a command not found :shrug:

Reply:
I can make it better though.